packages/odf/components/s3-browser/bucket-overview/BucketDetails.tsx

```diff
@@ -0,0 +1,144 @@
+      return t('SSE-S3 with KMS (Dual-layer)');
+    default:
+      return '';
+  }
+};
+
+const BucketOverview: React.FC<{}> = ({}) => {
+  const { t } = useCustomTranslation();
+  const { noobaaS3 } = React.useContext(NoobaaS3Context);
+  const { bucketName } = useParams();
+  const input = { Bucket: bucketName };
+  const { data: aclData } = useSWR(
+    `${bucketName}-${BUCKET_ACL_CACHE_KEY_SUFFIX}`,
+    () => noobaaS3.getBucketAcl(input)
+  );
+  const { data: encryptionData } = useSWR(
+    `${bucketName}-${BUCKET_ENCRYPTION_CACHE_KEY_SUFFIX}`,
+    () => noobaaS3.getBucketEncryption(input)
+  );
+  const { data: versioningData } = useSWR(
+    `${bucketName}-${BUCKET_TAGGING_CACHE_KEY_SUFFIX}`,
+    () => noobaaS3.getBucketVersioning(input)
+  );
+  const { data: tagData } = useSWR(
+    `${bucketName}-${BUCKET_VERSIONING_CACHE_KEY_SUFFIX}`,
+    () => noobaaS3.getBucketTagging(input)
+  );
+
+  // @TODO: review encryption description once the open questions are addressed.
+  const encryptionDescription = getEncryptionDescription(t, encryptionData);
+
+  const tags = tagData?.TagSet?.map((tag: Tag) => (
+    <Label className="pf-v5-u-mr-xs" color="grey" icon={<TagIcon />}>
+      {tag.Key}={tag.Value}
+    </Label>
+  ));
+
+  return (
+    <>
+      <div className="odf-m-pane__body">
+        <SectionHeading text={t('Bucket overview')} />
+        <div className="row">
+          <div className="col-sm-6">
+            <dl>
+              <dt>{t('Name')}</dt>
+              <dd>{bucketName}</dd>
+              <dt>{t('Tags')}</dt>
+              <dd>
+                <LabelGroup>{tags || DASH}</LabelGroup>
+              </dd>
+              <dt>{t('Owner')}</dt>
+              <dd>{aclData?.Owner?.DisplayName || DASH}</dd>
+            </dl>
+          </div>
+          <div className="col-sm-6">
+            <dl>
+              <dt>{t('Encryption')}</dt>
+              <dd>
+                {encryptionDescription ? (
+                  <>
+                    <LockIcon /> {t('Enabled')}
+                    <div className="pf-v5-u-disabled-color-100">
+                      {encryptionDescription}
+                    </div>
+                  </>
+                ) : (
+                  t('Disabled')
+                )}
+              </dd>
+              <dt>{t('Bucket versioning')}</dt>
+              <dd>
+                {versioningData?.Status === BucketVersioningStatus.Enabled ? (
+                  <>
+                    <GreenCheckCircleIcon /> {t('Enabled')}
+                  </>
+                ) : (
+                  t('Suspended')
+                )}
+              </dd>
+            </dl>
+          </div>
+        </div>
+      </div>
+    </>
+  );
+};
+
+type BucketDetailsProps = {
+  obj: K8sResourceKind;
+};
+
+export const BucketDetails: React.FC<BucketDetailsProps> = ({ obj }) => {
+  const { t } = useCustomTranslation();
+
+  return (
+    <>
+      <BucketOverview />
```

Comment:
nits (optional): can we rename it to just `Overview` or something else ?? There is a different (already existing) file/folder with exact same name, which would cause confusion.
```suggestion
      <Overview />
```

Reply:
Other option is to rename existing file/folder to a diff name.

Reply:
It has the title "Bucket overview" in the UX design, so I'll rename it to `BucketDetailsOverview`

Reply:
thanks